audit/audit.go

```diff
@@ -0,0 +1,72 @@
+package audit
+
+import (
+	"context"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// CryptoNewKeyEvent occurs when creating a new key.
+	CryptoNewKeyEvent = "CreateNewKey"
+	// CryptoSignJWTEvent occurs when signing a JWT.
+	CryptoSignJWTEvent = "SignJWT"
+	// CryptoSignJWSEvent occurs when signing a JWS.
+	CryptoSignJWSEvent = "SignJWS"
+	// CryptoDecryptEvent occurs when decrypting data.
+	CryptoDecryptEvent = "Decrypt"
+)
+
+// Info provides contextual information for auditable events.
+type Info struct {
+	// Actor is the user or service that performed the operation.
+	Actor string
+	// Operation is the name of the operation that was performed.
+	Operation string
+}
+
+// NewInfo formats audit info
+func NewInfo(actor string, operationModuleName string, operation string) Info {
+	return Info{
+		Actor:     actor,
+		Operation: operationModuleName + "." + operation,
+	}
+}
+
+var auditInfoContextKey = struct{}{}
```

Comment:
An empty struct could easily collide with types created by other packages/libs?

Reply:
That's right, I accidentally did this the wrong way. The correct way is;

```
type auditInfoContextKey struct{}

context.WithValue(ctx, auditInfoContextKey{}, value)
```